### ISSUE TITLE
Refactor: rename `n_swa` to `n_aux`

### ISSUE THREAD

**Issue by ngxson** [ggml-org/llama.cpp#13019]
### Background Description

Ref: https://github.com/ggml-org/llama.cpp/pull/12791#discussion_r2036666639

This won't be a simple renaming, but we also need to update some logic:

> I'll do this after having the logits matching automated test. The problem is that changing name `n_swa` to `n_pattern_aux` make the code checking `if (n_swa) then using SWA` becomes invalid. It should now become `if (n_pattern_aux && is_swa) then use SWA`
>
> I think it's better to add an enum called `llama_mask_aux` having 3 values: none, swa, chunked ; so that the code will become more clear

### Possible Refactor Approaches

Add `enum llama_mask_aux`

**Comment by ggerganov** [comment]
Btw, I was thinking a bit more about this, and SWA seems to fit both the regular SWA and the chunked Gemma-style attention. The latter is in some sense also "sliding", but in bigger "steps". So probably if we don't encounter more conceptually different use cases that make use of the `n_swa` functionality, it's OK to keep calling it like this.

**Comment by ngxson** [comment]
Hmm yeah that make sense then, let's keep `n_swa` then
